.github/workflows/check-lib.yml

```diff
@@ -0,0 +1,272 @@
+# Checks performed on our common library crates.
+name: Library
+
+on: [push, pull_request]
+
+env:
+  # Use `haswell` instead of `native` due to some GitHub Actions runners not
+  # supporting some `avx512` instructions.
+  RUSTFLAGS: -C target-cpu=haswell
+
+jobs:
+  # Run MSRV first to save Actions time if the code doesn't compile at all.
+  MSRV:
+    name: Minimum supported Rust version
+    runs-on: ubuntu-latest
+    env:
+      minrust: 1.53
+
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v2
+
+      - name: Install toolchain (${{ env.minrust }})
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: ${{ env.minrust }}
+          profile: minimal
+          override: true
+
+      - name: Cache dependencies
+        uses: Swatinem/rust-cache@v1
+
+      - name: Output processor info
+        run: cat /proc/cpuinfo
+
+      - run: cargo check --workspace --benches --examples --tests --all-features
+
+  # Check documentation
+  build-docs:
+    name: Build docs
+    needs: [MSRV]
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v2
+
+      - name: Install nightly toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: nightly
+          profile: minimal
+          override: true
+
+      - name: Build docs
+        env:
+          RUSTDOCFLAGS: --cfg docsrs -D broken_intra_doc_links
+        run: |
+          cargo doc --workspace --no-deps --features=permission-calculator
+          cargo doc -p twilight-util --no-deps --all-features
```

Comment:
This can be simplified (atleast drop nightly, `RUSTDOCFLAGS` and probably simplify to `cargo d --workspace --no-deps --all-features`) since the docs are only checked

Reply:
will this work cause of the `#[doc(cfg)]` attributes?

Reply:
we can't drop nightly because of `#[doc(cfg)]`, that's correct. see #813